sdk/forms/src/test/java/com/klaviyo/forms/KlaviyoWebViewDelegateTest.kt

```diff
@@ -0,0 +1,380 @@
+        verify { mockConfig.sdkVersion }
+        // tells us timer has started
+        assertEquals(staticClock.scheduledTasks.size, 1)
+    }
+
+    @Test
+    fun `show message type success case`() {
+        val delegate = KlaviyoWebViewDelegate()
+        delegate.initializeWebView()
+        delegate.postMessage("""{"type":"formWillAppear"}""")
+
+        verifyShow()
+    }
+
+    @Test
+    fun `malformed message throws an error`() {
+        val delegate = KlaviyoWebViewDelegate()
+        delegate.initializeWebView()
+        delegate.postMessage("""sawr a warewolf with a chinese menu inhis hands""")
+
+        verify {
+            spyLog.warning(
+                "Failed to relay webview message: sawr a warewolf with a chinese menu inhis hands",
+                any<JSONException>()
+            )
+        }
+    }
+
+    @Test
+    fun `deeplink but we have a null activity`() {
+        every { mockLifecycleMonitor.currentActivity } returns null
+
+        val deeplinkMessage = """
+            {
+              "type": "openDeepLink",
+              "data": {
+                "ios": "klaviyotest://settings",
+                "android": "klaviyotest://settings"
+              }
+            }
+        """.trimIndent()
+        val delegate = KlaviyoWebViewDelegate()
+        delegate.initializeWebView()
+        delegate.postMessage(deeplinkMessage)
+
+        verify {
+            spyLog.warning(
+                "Failed to launch deeplink klaviyotest://settings - null activity reference"
+            )
+        }
+    }
+
+    @Test
+    fun `appends asset source`() {
+        every { mockConfig.assetSource } returns "riders-on-the-stromboli"
+
+        val delegate = KlaviyoWebViewDelegate()
+        delegate.initializeWebView()
+
+        verify { spyLog.debug("Appending assetSource=riders-on-the-stromboli to klaviyo.js") }
+    }
+
+    @Test
+    fun `show message type null webview does not show`() {
+        val delegate = KlaviyoWebViewDelegate()
+        // notably do not init webview
+        delegate.postMessage("""{"type":"formWillAppear"}""")
+
+        verify { spyLog.warning("Unable to show IAF - null WebView reference") }
+        verifyShow(doesNotShow = true)
+    }
+
+    @Test
+    fun `show message type null activity`() {
+        every { mockActivity.window?.decorView } returns null
+
+        val delegate = KlaviyoWebViewDelegate()
+        delegate.initializeWebView()
+        delegate.postMessage("""{"type":"formWillAppear"}""")
+
+        verify { spyLog.warning("Unable to show IAF - null activity reference") }
+        verifyShow(doesNotShow = true)
+    }
+
+    @Test
+    fun `settings are properly set`() {
+        every {
+            anyConstructed<KlaviyoWebView>()
+                .settings
+        } returns mockSettings
+        val delegate = KlaviyoWebViewDelegate()
+        delegate.initializeWebView()
+
+        verify { mockSettings.javaScriptEnabled = true }
```

Comment:
Oh this is nice, does using the relaxed mock mean that a property actually gets set and saved in a map or something? I feel like setting this property in a strict mock would have thrown unless you explicitly mocked a setter?

Reply:
oh jeez i was mis reading this as `mockSettings.javaScriptEnabled == true`

Reply:
ok cool thx